src/mapInfoContributorManager.h

```diff
@@ -0,0 +1,52 @@
+/***************************************************************************
+ *   Copyright (C) 2021 by Piotr Wilczynski - delwing@gmail.com            *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#ifndef TMAPINFOCONTRIBUTORMANAGER_H
+#define TMAPINFOCONTRIBUTORMANAGER_H
+
+#include "pre_guard.h"
+#include <QMap>
+#include <QObject>
+#include <QString>
+#include "post_guard.h"
+
+#include "Host.h"
```

Comment:
As you are only referring to/using a pointer to the `Host` class in this header it is better to forward declare the use here **and move this line to the `.cpp` file**...
```suggestion
class Host;
```

Reply:
But in that case I get 
```mapInfoContributorManager.h:45:48: error: unknown type name 'Host'
[build]     MapInfoContributorManager(QObject* parent, Host* ph);```

Reply:
Any ideas @vadi2 - note my next comment (the one about) "You MUST include either a line containing Q_DECLARE_TR_FUNCTIONS(mapInfoContributorManager) or Q_OBJECT..."

Reply:
I will try to clean, haven't tried in that case.

Reply:
Still errors after clean.
Only way it works for me is when I do `#include`

Reply:
I'm not too familiar with forward declarations, so I'd stick with what works.

Reply:
:man_shrugging:

Well forward declarations (where a class is only needed for a pointer {or a reference} in a header file then it can be indicated with a `class ` declaration but the inclusion is left to the implementation {the `.cpp`} file) are generally a good idea but if it doesn't fly in this case - so be it...

Wikipedia: https://en.wikipedia.org/wiki/Forward_declaration#Classes
SE - several of the answers here: https://stackoverflow.com/q/4757565/4805858